--- rrdtool-1.7.0/src/rrd_graph.c	2017-05-16 12:26:46.000000000 +0000
+++ rrdtool-1.7.0/src/rrd_graph.c	2017-05-16 12:26:46.000000000 +0000
@@ -4300,27 +4300,6 @@
                                        im->ximg * im->zoom,
                                        im->yimg * im->zoom);
         break;
-#ifdef CAIRO_HAS_PDF_SURFACE
-    case IF_PDF:
-        im->gridfit = 0;
-        im->surface = im->graphfile
-            ? cairo_pdf_surface_create(im->graphfile, im->ximg * im->zoom,
-                                       im->yimg * im->zoom)
-            : cairo_pdf_surface_create_for_stream
-            (&cairo_output, im, im->ximg * im->zoom, im->yimg * im->zoom);
-        break;
-#endif
-#ifdef CAIRO_HAS_PS_SURFACE
-    case IF_EPS:
-        im->gridfit = 0;
-        im->surface = im->graphfile
-            ?
-            cairo_ps_surface_create(im->graphfile, im->ximg * im->zoom,
-                                    im->yimg * im->zoom)
-            : cairo_ps_surface_create_for_stream
-            (&cairo_output, im, im->ximg * im->zoom, im->yimg * im->zoom);
-        break;
-#endif
 #ifdef CAIRO_HAS_SVG_SURFACE
     case IF_SVG:
         im->gridfit = 0;
